### DISCUSSION TITLE
How to use a Vue slot with a template

### DISCUSSION THREAD

**Discussion by mikestu** [storybookjs/storybook#12691] [Help] [answered]
I'm using the general pattern that this guy uses [here](https://github.com/blowsie/storybook-vue/blob/master/src/stories/Button.stories.js).

What can I do if I need a new variation called "Secondary", which needs to specify markup for a slot in the my-button component?  Do I have to create a new template that has that slot configured, or is there some other way to accomplish that?

Example: 

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { MyButton },
  template: '<my-button @onClick="onClick" v-bind="$props" />',
});

export const Primary = Template.bind({});
Primary.args = {
  primary: true,
  label: 'Button',
};

**Comment by this-is-fine-dev** [comment]
up

**Comment by shodgson** [comment]
You can use the storybook props and use it within the slot, e.g.:

```js
const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { MyButton },
  template: '<my-button @OnClick="onClick" v-bind="$props" > {{ slotcontent }} </my-button>',
});

export const Primary = Template.bind({});
  Primary.args = {
  primary: true,
  label: 'Button',
};

export const Secondary = Template.bind({});
  Primary.args = {
  slotcontent: "Slot content here",
};
```

(I'm an absolute n00b on this, and found this question when I faced the same challenge. The above worked for me)

**Comment by euaaaio** [comment]
With Vue 3 and Storybook 6.2.0 beta we can make use slots as part of `args`:

```ts
import { Meta, Story } from '@storybook/vue3'
import { defineComponent } from 'vue'

import Badge from './Badge.vue'

export default {
  title: 'Badge',
  component: Badge,
  argTypes: {
    default: {
      control: 'text',
      description: 'Slot content',
      defaultValue: 'Badge'
    }
  }
} as Meta

export const Default: Story = args => defineComponent({
  components: { Badge },
  setup: () => ({ args }),
  template: '<p-badge>{{ args.default }}</p-badge>'
})
```

And even change slot’ content:

```ts
export const AnotherText = Template.bind({})
AnotherText.args = {
  default: 'Another text'
}
```

**Reply by vchan14** [reply]
Can we pass HTML elements to the slot value? 
i.e:` '<p> Anothertext</p>' `
I tried it and it doesn't evaluate the paragraph tag and literally displayed the exact string.

**Reply by euaaaio** [reply]
Perhaps `v-html` is what you need.

**Reply by mryellow** [reply]
This is fine up until you wish to pass actual slot properties.

**Comment by tylercollier** [comment, marked as answer]
Here's how I did it in Vue 2. I'm using this example button component with a default slot and an optional "hint" slot.

Button.vue:
```vue
<template>
  <button @click="$emit('onClick')">
    <slot>
      Submit
    </slot>
    <div style="font-size:x-small;color:gray;">
      <slot name="hint">
        Click me
      </slot>
    </div>
  </button>
</template>

<script>
export default {
}
</script>
```

Button.stories.js:
```js
const Template = (args, { argTypes }) => {
  return {
    props: Object.keys(argTypes),
    components: { Button },
    template:
      `
      <Button @onClick="onClick" v-bind="$props">
        <template v-if="${'default' in args}" v-slot>${args.default}</template>
        <template v-if="${'hint' in args}" v-slot:hint>${args.hint}</template>
      </Button>
    `,
  }
}

export const Base = Template.bind({})
Base.args = {}

export const WithDefaultSlotContent = Template.bind({})
WithDefaultSlotContent.args = {
  default: `<strong>Go!</strong>`,
}

export const WithOptionalSlotContent = Template.bind({})
WithOptionalSlotContent.args = {
  hint: `<div style="color:red;">Hurry</div>`,
}
```

You can see we're using `<template v-if="...">` within the Template, so that the slots can be optional. With this approach, the `WithDefaultSlotContent` story will actually have a bold `Go!`. I was originally trying `{{$props.default}}` and it would put the literal markup `<strong>Go!</strong>`.

**Reply by mikestu** [reply]
That’s clever, thanks!

**Reply by thiagonunesbatista** [reply]
I saw the name of the slot is being showed as a prop in the code preview of the story, is that the correct waty of documenting this?

I tried to use a Custom component in a slot but It didn't work because the content was just a string

**Reply by tylercollier** [reply]
@thiagonunesbatista Sorry, I haven't used the documenting feature much. Where Storybook shows Props under the story, I also see a section named Slots, and in it, I see `default` and `hint`.

I was able to use a Custom component in a slot just fine. Perhaps you can open a separate discussion?

**Reply by thiagonunesbatista** [reply]
![Screenshot from 2021-04-01 09-29-48](https://user-images.githubusercontent.com/67487679/113293988-d5083280-92cc-11eb-954e-459e4b058f8b.png)

I attatched a screenshot, the problem is in this part, when I try to see the code preview using this approach, the slot "text" is appearing as a prop, I'm trying to find a way that shows the usage of the slot in this code preview.

**Reply by bkuzma** [reply]
Hi there, I'm curious if you've found a way to be able to edit the slot content via a slot "control" at the bottom of the story? I get a `-` in place of where the control should be (see first screenshot). 

I've tried the approach of defining the control inside of `argTypes` in the default export (code snippet below), which did give me a text input, but when I type a value into that input, it's not reflected in the component (see second screenshot).

EDIT: I went to the docs tab of my component and when I typed into the control for my slot there, the component **did** get updated with the value, and then when I clicked back into the Canvas view, the new value was reflected there as well. But typing into the slot control in the Canvas view still doesn't update the component. Storybook bug?

```
export default {
    title: 'VCheckbox',
    component: VCheckbox,
    argTypes: {
        label: {
            // not a prop!
            control: {
                type: 'text',
            },
            defaultValue: 'A Button Label',
            meta_description: 'slot content',
            table: {
                type: {
                    summary: null,
                },
            },
        },
    },
};
```

![image](https://user-images.githubusercontent.com/448483/118848206-d28a8800-b8ce-11eb-85fd-23e9d25fb00d.png)
![image](https://user-images.githubusercontent.com/448483/118848654-3b720000-b8cf-11eb-8db4-a79f0b99e5ed.png)

**Reply by tylercollier** [reply]
@bkuzma I just observed the behavior as you. If I edit the value in the Canvas, it doesn't update until I switch to the Docs, where it always works. Yeah, I'm guessing a bug in storybook.

**Reply by skstuder** [reply]
This answer seems to answer how to use a named slot, but what about a default slot. Following a similar path with the use of 'default' does not lead to the desired result. (vue, jsx)

below code updates control to text box which is expected: 
```
argTypes = {{ 
    default: {
        control: {
            type:'text'
        }
    }
}}
```
![Screen Shot 2021-05-24 at 7 07 09 AM](https://user-images.githubusercontent.com/55567604/119359761-cac83c00-bc5e-11eb-8dba-1865147a1027.png)

but trying to connect the value using the same default name does nothing. 
```
args = {{
    default: 'example text' //does not render in canvas component
}}
```
![Screen Shot 2021-05-24 at 7 07 50 AM](https://user-images.githubusercontent.com/55567604/119359783-d0be1d00-bc5e-11eb-87d8-ae3df2ff897e.png)

**Reply by wouterkroes** [reply]
> @bkuzma I just observed the behavior as you. If I edit the value in the Canvas, it doesn't update until I switch to the Docs, where it always works. Yeah, I'm guessing a bug in storybook.

Is there al ready a bug report of this?

**Reply by bkuzma** [reply]
I'm not sure, sorry.

**Reply by wouterkroes** [reply]
Could not find it so I reported it (https://github.com/storybookjs/storybook/issues/16750)

**Reply by csalvato** [reply]
When using this approach, does the `Show Code` section in docs work for you all? I am not seeing my slot content in the code section 🤔 

Here's my template:

```js
template: `<SplitActionDropdown
      :dropdown-classes="['flow-left']"
      v-bind="filteredProps"
      @button-click="onButtonClick"
      @split-button-click="onSplitButtonClick"
    >
      ${args.slotContent}
    </SplitActionDropdown>
    `,
```

<img width="448" alt="Screen Shot 2021-12-16 at 1 38 37 PM" src="https://user-images.githubusercontent.com/1158152/146445650-8f5605c3-8830-4a7f-aa40-9b11bc7b1ce8.png">

I would expect to see this:

```vue
<template>
  <SplitActionDropdown
    id="example-split-action-drop1"
    buttonIcon="icons/edit"
    buttonLabel="Edit - Left Aligned Dropdown"
  >
    <template #dropdownListItems>
      <li><a>Copy</a></li>
      <li><a>Delete</a></li>
    </template>
  </SplitActionDropdown>
</template>
```

IIUC, it seems that slots in Vue 2 + Storybook are just broken? Is there better support with Vue 3?

**Reply by tylercollier** [reply]
I'm not sure @csalvato, sorry!

**Reply by braddialpad** [reply]
> IIUC, it seems that slots in Vue 2 + Storybook are just broken? Is there better support with Vue 3?

We have been struggling with this issue for ages, and yes slots seem to basically just be broken in storybook code examples, it seems like storybook does not give much of a shit about Vue TBH. Looked into vue 3 a bit and from what I have seen it is not fixed there either, seems anecdotally it may be even worse?

The only way I could find to get slots to display correctly in the docs is if you use the deprecated `slot=""` syntax, however it seems to break if you use any scoped slots. Example below:

```js
const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { MyComponent },
  template: `
<my-component v-bind="$props">
  <template v-if="${'anchor' in args}" slot="anchor">${args.anchor}</template>
  <template v-if="${'content' in args}" slot="content">${args.content}</template>
</my-component>`,
});
```

**Reply by zhixuan-loh** [reply]
> > IIUC, it seems that slots in Vue 2 + Storybook are just broken? Is there better support with Vue 3?
> 
> We have been struggling with this issue for ages, and yes slots seem to basically just be broken in storybook code examples, it seems like storybook does not give much of a shit about Vue TBH. Looked into vue 3 a bit and from what I have seen it is not fixed there either, seems anecdotally it may be even worse?
> 
> The only way I could find to get slots to display correctly in the docs is if you use the deprecated `slot=""` syntax, however it seems to break if you use any scoped slots. Example below:
> 
> ```js
> const Template = (args, { argTypes }) => ({
>   props: Object.keys(argTypes),
>   components: { MyComponent },
>   template: `
> <my-component v-bind="$props">
>   <template v-if="${'anchor' in args}" slot="anchor">${args.anchor}</template>
>   <template v-if="${'content' in args}" slot="content">${args.content}</template>
> </my-component>`,
> });
> ```

Hi I faced the same issue as you but I solved it. 

I'm using Vue 3 & `"@storybook/vue3": "^6.4.19"`. 

Instead of `slot="slot-name"`, use `v-slot:slot-name`. Both Vue and Storybook accepted it.


Cheers.

**Reply by yonario** [reply]
This is in the official docs
https://storybook.js.org/docs/vue/writing-stories/args#args-can-modify-any-aspect-of-your-component
Apparently you can do `v-if="args.default"` instead of `v-if="${'default' in args}"` without the need for breaking out of the template.

**Reply by younho9** [reply]
> > IIUC, it seems that slots in Vue 2 + Storybook are just broken? Is there better support with Vue 3?
> 
> We have been struggling with this issue for ages, and yes slots seem to basically just be broken in storybook code examples, it seems like storybook does not give much of a shit about Vue TBH. Looked into vue 3 a bit and from what I have seen it is not fixed there either, seems anecdotally it may be even worse?
> 
> The only way I could find to get slots to display correctly in the docs is if you use the deprecated `slot=""` syntax, however it seems to break if you use any scoped slots. Example below:
> 
> ```js
> const Template = (args, { argTypes }) => ({
>   props: Object.keys(argTypes),
>   components: { MyComponent },
>   template: `
> <my-component v-bind="$props">
>   <template v-if="${'anchor' in args}" slot="anchor">${args.anchor}</template>
>   <template v-if="${'content' in args}" slot="content">${args.content}</template>
> </my-component>`,
> });
> ```

In Vue2, because [Named slots with the `slot` attribute is deprecated](https://v2.vuejs.org/v2/guide/components-slots.html#Named-Slots-with-the-slot-Attribute), [`transformSource`](https://github.com/storybookjs/storybook/blob/7035ea7389393da041985ebc491ee58dedb50d06/code/addons/docs/docs/recipes.md#customizing-source-snippets) can be used to transform to [updated named slots syntax](https://v2.vuejs.org/v2/guide/components-slots.html#Named-Slots)

```js
// preview.js
const parameters = {
	docs: {
		transformSource(source: string) {
			return source.replace(/slot="([^"]*)"/g, (_, p1) => `v-slot:${p1}`);
		},
	}
}
```

![image](https://user-images.githubusercontent.com/48426991/180958861-93a4a3af-edb3-49ce-91ad-cbd1c027780d.png)

**Comment by camslice** [comment]
Not directly answering the question here, but providing something useful which is related to the issue. If you define your templates using string literal syntax, then you can grab the template string from `[template-name].template`

Button.vue
```vue
<template>
  <button>
    <slot/>
  </button>
</template>
```

Foo.js
```js
export default {
  template: `<p>Custom foo</p>`
};
```

Bar.js
```js
export default {
  template: `<p>Custom bar</p>`
};
```

Button.stories.js
```js
import Button from './Button.vue';
import Foo from './Foo.js';
import Bar from './Bar.js';

const Template = (args, { argTypes }) => {
  return {
    props: Object.keys(argTypes),
    components: { Button },
    template:
      `
      <Button @onClick="onClick" v-bind="$props">
        ${args.slotTemplate}
      </Button>
    `,
  }
}

export const Foo = Template.bind({})
Foo.args = {
  slotTemplate: Foo.template
}

export const Bar = Template.bind({})
Bar.args = {
  slotTemplate: Bar.template
}
```

**Comment by yvettelexter** [comment]
<template>
  <button @click="$emit('onClick')">
    <slot>
      Submit
    </slot>
    <div style="font-size:x-small;color:gray;">
      <slot name="hint">
        Click me
      </slot>
    </div>
  </button>
</template>

<script>
export default {
}
</script>

**Comment by BryanAdamss** [comment]
I got a new way use `slot` with `mdx`.

It use `"@storybook/vue": "^6.2.8"`；

The complete example can be referred to `https://github.com/BryanAdamss/monorepo-starter` ; 
Its a mono repo with `storybook` and `jsdoc` for `vue2 compoent` and `lib` development.

```mdx
<!-- args定义参考:https://storybook.js.org/docs/vue/writing-stories/args -->
<!-- argTypes定义参考:https://storybook.js.org/docs/react/essentials/controls#annotation -->

import {
  Meta,
  Canvas,
  Story,
  ArgsTable,
  Description
} from '@storybook/addon-docs/blocks'
import BaButton from '../src/button.vue'

<Meta
  title="Components/BaButton"
  component={BaButton}
  argTypes={{
    backgroundColor: { control: 'color' },
    size: { control: { type: 'select', options: ['small', 'medium', 'large'] } }
  }}
/>

<!-- 创建容器组件 -->

export const Template = (args, storyCtx) => ({
  // args包含storybook的所有参数，包含props、slots、listeners
  // storyCtx除了包含args外，还包含更多额外信息
  props: Object.keys(args), // 接收所有args上的参数，并将其定义为props，直接通过this可访问
  mounted() {
    console.log('storyCtx', storyCtx)
    // 修复component name错误
    const {
      parameters: {
        component: { name }
      }
    } = storyCtx
    this._vnode.tag = name || 'BaButton'
  },
  render(h) {
    return h(BaButton, {
      props: this.$props,
      scopedSlots: {
        // 默认插槽
        default: (props) => {
          // default通过args绑定到props上，所以可以通过this.props直接访问
          return this.default ? h(this.default, { props }) : void 0
        },
        // 具名插槽
        icon: (props) => {
          return this.icon ? h(this.icon, { props }) : void 0
        }
      }
    })
  }
})

# BaButton

<!-- 读取组件描述 -->

<Description of={BaButton} />

<Canvas>
  <Story
    name="Primary"
    args={{
      label: '按钮'
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable of={BaButton} />

## 尺寸

可以通过设置`size`控制按钮大小；可选值有`['small', 'medium', 'large']`；默认`medium`

<Canvas>
  <Story
    name="Large"
    args={{
      size: 'large',
      label: '大小'
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## 颜色

可通过设置`backgroundColor`控制背景颜色；支持`hex、rgb、rgba、关键字`

<Canvas>
  <Story
    name="BgColor"
    args={{
      backgroundColor: 'red',
      label: '颜色'
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## slot:icon

可通过`icon`插槽自定义图标样式

<!-- 由于story自动生成的code样例无法展示slot，建议通过docs.source.code手动定义 -->

<Canvas>
  <Story
    name="SlotIcon"
    args={{
      label: '按钮',
      icon: {
        // 以template形式声明组件
        template: '<span>x</span>'
      }
    }}
    parameters={{
      docs: {
        source: {
          code: `
<template>
  <BaButton>
    <template #icon>
      <span>x</span>
    </template>
  </BaButton>
</template>`
        }
      }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```

**Reply by BryanAdamss** [reply]
It works like this 
![image](https://user-images.githubusercontent.com/7441504/118209595-7d4c1200-b49b-11eb-8141-24f1f88a986e.png)

**Comment by paulmelero** [comment]
Here's how I am doing it. It's a mix between @camslice's answer and [vue-kitchen-sink](https://raw.githubusercontent.com/storybookjs/storybook/201ae32/examples/vue-kitchen-sink/src/stories/addon-controls.stories.mdx)'s example:

Imagine you have this component:

```vue
<template>
  <button type="button">
    <slot />
  </button>
</template>
```

And this is how currently our stories in MDX look like:

```mdx
import { Meta, Canvas, Story } from '@storybook/addon-docs'
import MyButton from './Button.vue'

<Meta title="Button" component={MyButton} />

export const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { MyButton },
  template: `
    <MyButton v-bind="filteredProps">
      ${args.slotTemplate}
    </MyButton>`,
  computed: {
    /*
      This excludes "slotTemplate" from props, so it's not shown
      on the "Show code" block as a prop.
     */
    filteredProps() {
      const { slotTemplate, ...props } = this.$props
      return props
    },
  },
})

# Button

These are our buttons:

<Canvas>
  <Story
    name="Basic"
    args={{
      slotTemplate: `Basic`,
    }}
  >
    {Template.bind({})}
  </Story>
  <Story
    name="Nested HTML"
    args={{
      slotTemplate: `<span>Some text <em> with nested <strong>HTML</strong></em></span>`,
    }}
  >
    {Template.bind({})}
  </Story>
  <Story
    name="Disabled"
    args={{
      slotTemplate: `Can't click me :(`,
      disabled: true,
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>
```

This is the result: The component's code is wrapped in a `<template>`, but that's a fair trade-off for us:

![image](https://user-images.githubusercontent.com/6775220/125023240-b7dac100-e07e-11eb-93eb-0f61015b31a3.png)

## 💡 
- For an **alternative** solution, using `argTypes` + `v-slot`, instead of `args`, check this article: https://highlandsolutions.com/blog/stop-designing-components-in-your-vue-application

**Comment by anhance** [comment]
This solution works for now, but as mentioned above in comments, you can edit only in "Docs"

```
import BaseButton from './BaseButton.vue'

export default {
  title: 'UI/button/BaseButton',
  component: BaseButton,
  argTypes: {
    default: {
      control: {
        type: 'text'
      },
      defaultValue: 'Button'
    },
    label: {
      control: {
        type: 'text'
      },
      defaultValue: 'Label'
    }
  }
}

export const BaseButtonComponent = (args, { argTypes }) => ({
  components: { BaseButton },
  props: Object.keys(argTypes),
  template: `
    <base-button v-bind="$props">
      <template v-if="${'default' in args}">${args.default}</template>
      <template v-if="${'label' in args}" v-slot:label>${args.label}</template>
    </base-button>
  `
})
```

**Comment by Davo911** [comment]
Only way I got it working as I wanted to was with `v-html`:

```typescript
const Template = (args: any) => ({
  components: { ... },
  setup() {
    return { args };
  },
  template: `<widget v-bind="args" > ` +
    `<template v-if="${'slotLeft' in args}"   #left><div v-html="args.slotLeft"/></template> ` +
    `<template v-if="${'slotRight' in args}"  #right><div v-html="args.slotRight"/></template> ` +
    `<template v-if="${'slotBottom' in args}" #bottom><div v-html="args.slotBottom"/></template> ` +
    ' </dt-widget-header>',
});

export const Default: Story = Template.bind({});
Default.args = {
  //PROPS
  ...,

  //SLOTS
  slotLeft: '<span>LEFT SLOT</span>',
  slotRight: '<span>RIGHT SLOT</span>',
  slotBottom: '<span>BOTTOM SLOT</span>',

}
```

**Reply by Christo44** [reply]
Good idea, here's another similar way:
```
<widget>
<template v-if="${'suffix' in args}" #suffix>
    {{args.suffix}}
  </template>
</widget>
```

**Comment by yooouuri** [comment]
Another solution

```typescript
import { Meta, Story } from '@storybook/vue3'
import { defineComponent, h } from 'vue'

...

const Template: Story<typeof Component> = (args) =>
  defineComponent({
    components: {
      Component,
    },
    setup() {
      const slots = Object.keys(args.slots).reduce((prev, curr) => {
        if (typeof args.slots[curr] === 'string') {
          return { ...prev, ...{ [curr]: () => args.slots[curr] } }
        }

        return { ...prev, ...{ [curr]: () => h(args.slots[curr]) } }
      }, {})

      return function () {
        return h(
          Component,
          {
            ...args.props,
          },
          slots
        )
      }
    },
  })

export const Default = Template.bind({})
Default.args = {
  props: {
    search: true,
    searchPlaceholder: 'Zoek product',
    addButton: true,
  },
  slots: {
    button: 'Nieuw product1', // string is a valid VNode
    sort: { // everything else lets wrap it into a h
      data() {
        return {
          test: '1',
        }
      },
      template: `{{ test }}`,
    },
  },
}
```

**Comment by luckylooke** [comment]
I have simplified template for multislots components.

```html
<my-button v-bind="args">
        <template v-for="[name, content] in args.slots" #[name]>
            {{ content }}
        </template>
</my-button>
```

then you have several options how to fill in slots

```js
// PrependSlot.args = { slots: Object.entries({ prepend: '#' })};
// PrependSlot.args = { slots: new Map([['prepend', '#']])};
PrependSlot.args = { slots: [['prepend', '# ']]};
// etc..
```

Happy coding :)

**Comment by avivday2** [comment]
Best way I found:

```

export const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes).filter((key) => {
    const excludedSlots = ["default"]; /** define all of the excluded slots (or even props if you want)*/
    return !excludedSlots.includes(key); /** exclude ** /
  }),
  components: { MyButton },
  template: `<MyButton v-bind="$props" @click="click" @contextmenu="contextmenu">
      ${args.default}
  </MyButton>`,
});
```

**Comment by sherif414** [comment]
is this fixed in storybook 7? 
i tried some of the solutions above but they don't work well with typescript

**Reply by Davo911** [reply]
I can confirm, that the v-html version works finde with typescript

**Reply by sherif414** [reply]
i can't change the form control element from object to text without getting typescript errors
` argTypes: {
    size: { control: 'select', options: ['sm', 'md', 'lg'], defaultValue: 'md' },
    default: {control: 'text'}
  }, `
here typescript complains Object literal may only specify known properties, and 'default' does not exist in type

**Reply by matthew-dean** [reply]
Did anyone get this working with TypeScript?

**Reply by JoJk0** [reply]
I'm working at wrapper type that I can wrap `typeof MyComponent` to append slots as props to the component type.

**Reply by JoJk0** [reply]
```ts
export type SlottedMeta<TComponent extends abstract new (...args: any) => any, TSlots extends string> = Meta<DefineComponent<InstanceType<TComponent>['$props'] & Record<TSlots, string>>>
```

Use it instead of Meta like `... } satisfies SlottedMeta<typeof MyComponent, 'default'|'mySlot'>`

**Comment by ahmadertarizqi** [comment]
Hi, is this discussion still active? Currently, I am using storybook 9.1.15 and vue 3 + typescript.
When I tried several solutions on this forum and read the instruction in official documentation to render vue component slots, it didn't seem to work properly. Only working for html or string, not vue component.
Is there any solution for the version I am using?

Official docs: https://storybook.js.org/docs/writing-stories/args#args-can-modify-any-aspect-of-your-component
